src/compositions/useRouter.ts

```diff
@@ -0,0 +1,14 @@
+import { inject } from 'vue'
+import { Router } from '@/types/router'
+import { routerInjectionKey } from '@/utilities'
+
+export function useRouter(): Router {
+  const router = inject(routerInjectionKey)
+
+  if (!router) {
+    throw new Error('Router is not installed')
+  }
+
+  // any prevents typescript from trying to verify the types are compatible
+  return router as any
```

Comment:
do we actually need this? In my testing locally it works fine without `as any`

Reply:
I was getting infinitely deep errors. Possible the cause of that was fixed. Will check

Reply:
Yep no longer needed. Good catch 👍